opm/material/common/UniformXTabulated2DFunction.hpp

```diff
@@ -368,11 +378,17 @@ class UniformXTabulated2DFunction
         auto yLower =  y - alpha*shift;
         auto yUpper =  y + (1-alpha)*shift;
 
-        unsigned j1 = ySegmentIndex(yLower, i, extrapolate);
-        unsigned j2 = ySegmentIndex(yUpper, i + 1, extrapolate);
-        const Evaluation& beta1 = yToBeta(yLower, i, j1);
-        const Evaluation& beta2 = yToBeta(yUpper, i + 1, j2);
+        j1 = ySegmentIndex(yLower, i, extrapolate);
+        j2 = ySegmentIndex(yUpper, i + 1, extrapolate);
+        beta1 = yToBeta(yLower, i, j1);
+        beta2 = yToBeta(yUpper, i + 1, j2);
 
+    }
+
+    template <class Evaluation>
+    Evaluation eval(unsigned& i, unsigned& j1, unsigned& j2, Evaluation& alpha, Evaluation& beta1, Evaluation& beta2,
```

Comment:
Do not pass `i`, `j1`, `j2`, `alpha`, `beta1` or `beta2` as mutable references (the integers as values, the evaluations as const references). Do not pass the `x`, `y` or `extrapolate` at all.

Reply:
done